net/dns/manager.go

```diff
@@ -67,12 +68,19 @@ func NewManager(logf logger.Logf, oscfg OSConfigurator, health *health.Tracker,
 	if goos == "" {
 		goos = runtime.GOOS
 	}
+
+	disableSplitDNSOptimization := false
+	if knobs != nil {
+		disableSplitDNSOptimization = knobs.DisableSplitDNSWhenNoCustomResolvers.Load()
```

Comment:
i think you probably want to pass in `knobs` otherwise this will only read the value at construction time which might be too early, like even before a netmap.

Reply:
done!